src/main/java/com/scottlogic/context/FilterContext.java

```diff
@@ -0,0 +1,16 @@
+package com.scottlogic.context;
+
+import com.scottlogic.UserPost;
+import com.scottlogic.strategy.FilterStrategy;
+
+import java.util.List;
+
+public class FilterContext {
```

Comment:
You don't really need a class like this.

The context in the strategy pattern is whatever uses your strategy, so like an ATM.
Here you can just use it in your main or in your test.

Reply:
Does this mean I'll just be calling the classes directly? as the Context file is used as a constructor for the desired filter

Reply:
Yep, this might be a bit more advanced for now...
Imagine a context like ATM, you can declare that it can filter with an interface.
A design pattern is to have an ATMFactory class. It can build your ATM with a specific filter (like filterAuthor) you ask for.
Then through dependency injection, your factory can create your ATM passing the concrete strategies in the constructor or via a setter.

ATM knows nothing about AuthorFilter, but it knows it can call filter.

Maybe you're trying to do factory pattern here, but I'd allow a simpler approach for now.

Reply:
For the time being I've just deleted the Context class here and am calling the classes directly - fixed